extensions/cl_ext_cxx_for_opencl.asciidoc

```diff
@@ -0,0 +1,117 @@
+// Copyright 2018-2020 The Khronos Group. This work is licensed under a
+// Creative Commons Attribution 4.0 International License; see
+// http://creativecommons.org/licenses/by/4.0/
+
+:data-uri:
+:icons: font
+include::../config/attribs.txt[]
+:source-highlighter: coderay
+
+= cl_ext_cxx_for_opencl
+:R: pass:q,r[^(R)^]
+Khronos{R} OpenCL Working Group
+
+== Name Strings
+
+`cl_ext_cxx_for_opencl`
+
+== Contact
+
+Please see the *Issues* list in the Khronos *OpenCL-Docs* repository: +
+https://github.com/KhronosGroup/OpenCL-Docs
+
+== Contributors
+
+Kevin Petit, Arm Ltd. +
+Sven Van Haastregt, Arm Ltd. +
+Anastasia Stulova, Arm Ltd. +
+Marco Antognini, Arm Ltd. +
+Neil Hickey, Arm Ltd. +
+
+== Notice
+
+include::../copyrights.txt[]
+
+== Version
+
+Built On: {docdate} +
+Version: 1.0.0
+
+== Dependencies
+
+This extension is written against the OpenCL Specification
+Version 3.0.2.
+
+This extension requires OpenCL 3.0 or OpenCL 2.x and `cl_khr_extended_versioning`.
```

Comment:
> This extension requires OpenCL 3.0 or **OpenCL 2.x** and `cl_khr_extended_versioning`.

Emphasis mine.  Mostly out of curiosity rather than because I have a need for it, but is there any reason why OpenCL 1.2 and `cl_khr_extended_versioning` wouldn't be possible?

Reply:
C++ for OpenCL depends on OpenCL C 2.0 at the moment. We haven't really looked at how it interacts with OpenCL C 3.0 and optional language features. I'll add an explicit requirement on OpenCL C 2.0 here.